src/test/java/project/service/GasStationServiceImplTest.java

```diff
@@ -18,76 +18,128 @@
 import java.math.RoundingMode;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.NoSuchElementException;
 
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
 
 @ExtendWith(MockitoExtension.class)
 class GasStationServiceImplTest {
-
     private GasStationServiceImpl gasStationService;
-    private final ModelMapper modelMapper = new ModelMapper();
     @Mock
     private GasStationRepository repository;
+    private final ModelMapper modelMapper = new ModelMapper();
 
     @BeforeEach
-    void setUp() {
-        gasStationService = new GasStationServiceImpl(repository, modelMapper);
-
+    void setup() {
+        this.gasStationService = new GasStationServiceImpl(repository, modelMapper);
     }
+
     @Test
-    void getStationsByName() {
+    void getStationsByNameTest() {
+
+        // GIVEN
         GasStation testGasStation = createTestGasStation();
         when(repository.findByName(testGasStation.getName())).thenReturn(List.of(testGasStation));
+
+        // WHEN
         List<GasStationDTO> stationsByName = gasStationService.getStationsByName("TestName");
-        Assertions.assertEquals(1, stationsByName.size());
-        Assertions.assertEquals(testGasStation.getName(), stationsByName.get(0).getName());
+
+        // THEN
+        assertEquals(1, stationsByName.size());
+        assertEquals(testGasStation.getName(), stationsByName.get(0).getName());
+    }
+
+    @Test
+    void getStationsByNameTestWithInvalidName() {
+        // GIVEN WHEN
```

Comment:
here you can just keep the `// WHEN`

Reply:
GIVEN and WHEN are split to GIVEN and WHEN THEN